packages/scm/src/browser/scm-service.ts

```diff
@@ -0,0 +1,325 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { Disposable, DisposableCollection, Emitter, Event } from '@theia/core/lib/common';
+import { injectable } from 'inversify';
+import URI from '@theia/core/lib/common/uri';
+
+export const ScmService = Symbol('ScmService');
+export interface ScmService extends Disposable {
+
+    readonly onDidAddRepository: Event<ScmRepository>;
+    readonly onDidRemoveRepository: Event<ScmRepository>;
+
+    readonly repositories: ScmRepository[];
+    selectedRepository: ScmRepository | undefined;
+    readonly onDidChangeSelectedRepositories: Event<ScmRepository>;
+
+    registerScmProvider(provider: ScmProvider): ScmRepository;
+}
+
+export interface ScmProvider extends Disposable {
+    readonly label: string;
+    readonly id: string;
+    readonly handle: number;
+    readonly contextValue: string;
+
+    readonly groups: ScmResourceGroup[];
+
+    readonly onDidChangeResources: Event<void>;
+
+    readonly rootUri?: string;
+    readonly count?: number;
+    readonly commitTemplate?: string;
+    readonly onDidChangeCommitTemplate?: Event<string>;
+    readonly onDidChangeStatusBarCommands?: Event<ScmCommand[]>;
+    readonly acceptInputCommand?: ScmCommand;
+    readonly statusBarCommands?: ScmCommand[];
```

Comment:
`ScmProvider.statusBarCommands` is not used in scm extension.

Reply:
removed unused keys

Reply:
removed unused keys